zeebe/benchmarks/project/src/main/java/io/camunda/zeebe/Worker.java

```diff
@@ -87,6 +81,85 @@ public void run() {
                 }));
   }
 
+  private JobHandler handleJob(
+      final ZeebeClient client,
+      final String variables,
+      final long completionDelay,
+      final BlockingQueue<Future<?>> requestFutures) {
+    return (jobClient, job) -> {
+      // we record the start handling time to better calculate the completion delay
+      // as when we send a message we already have a delay due to waiting on the response
+      final long startHandlingTime = System.currentTimeMillis();
+
+      if (workerCfg.isSendMessage()) {
+
+        final var correlationKey =
+            job.getVariable(workerCfg.getCorrelationKeyVariableName()).toString();
+
+        final boolean messagePublishedSuccessfully = publishMessage(client, correlationKey);
+        if (!messagePublishedSuccessfully) {
+          // - On issues with publishing a message we need to retry the job
+          // - thus is to make sure our message gets published
+          // - otherwise our process might get stuck
+          // - failing the job makes it immediately available again
+          jobClient
+              .newFailCommand(job)
+              .retries(job.getRetries())
+              .errorMessage("Message publish failed.")
+              .send();
+          return;
```

Comment:
:question: Does it have to be blocking? Can't we use

```
 publishMessageFuture.whenComplete(ok, error) -> {
  if(error) 
     fail  job
  else
     complete job
 }
 ```

Reply:
I was not sure as it would change the behavior, especially where the delay is executed etc